### ISSUE TITLE
Yolov3_5l map hangs

### ISSUE THREAD

**Issue by AdamCuellar** [AlexeyAB/darknet#5252]
Using the map command, yolov3_5l model hangs at the following:

![image](https://user-images.githubusercontent.com/40183888/79578054-b09f5000-8093-11ea-9b61-d973a905d6f4.png)

However, the regular yolov3 model works fine on the same dataset. Not sure why its getting stuck. Attached is my cfg file.

[yolov3_5l_obj.cfg.txt](https://github.com/AlexeyAB/darknet/files/4493432/yolov3_5l_obj.cfg.txt)

**Comment by AlexeyAB** [comment]
Try to train more than 1000 iterations (4000 iterations) before mAP calculation.

**Comment by AdamCuellar** [comment]
@AlexeyAB Okay will do. I was trying to see how the training was going since yolov3 got an unfavorable mAP of 28%. 
I have objects in images with width=650 height=512 and the following object stats:
area minimum = 3
area maximum = 2400
area mean = 369
area std = 479

Is yolov3_5l the best model to use for this?

**Comment by AlexeyAB** [comment]
* Do you train both yolov3.cfg and yolov3_5l.cfg with `width=640 height=512` in cfg-file?

* Did you train to train yolov3.cfg with `width=640 height=512` and with this suggestion? https://github.com/AlexeyAB/darknet#how-to-improve-object-detection


> or training for small objects (smaller than 16x16 after the image is resized to 416x416) - set layers = -1, 11 instead of https://github.com/AlexeyAB/darknet/blob/6390a5a2ab61a0bdf6f1a9a6b4a739c16b36e0d7/cfg/yolov3.cfg#L720 and set stride=4 instead of https://github.com/AlexeyAB/darknet/blob/6390a5a2ab61a0bdf6f1a9a6b4a739c16b36e0d7/cfg/yolov3.cfg#L717

**Comment by AdamCuellar** [comment]
@AlexeyAB A while back I trained a keras-yolov3 model with a subset of this data at 512x512 and got a ~70% mAP. 

Then I trained the whole dataset with yolov3 (with this repo) with 640x512 with no other changes to the cfg and got the 28% mAP. 

I’ve tried the small object suggestions mentioned above but along with other changes so I’ll try those changes alone and report back with the performance.

For yolov3_5l, I just started the training with the previously attached cfg last night. I stopped it to evaluate how it was doing. But after I try yolov3 with your suggested changes I’ll train again.

A side question: I have a titan RTX, what’s the recommended batch size and subdivisions?

**Comment by AlexeyAB** [comment]
batch=64
subdivisions=8 or 4 or 2 or 1 as low as possible

> A while back I trained a keras-yolov3 model with a subset of this data at 512x512 and got a ~70% mAP.
> Then I trained the whole dataset with yolov3 (with this repo) with 640x512 with no other changes to the cfg and got the 28% mAP.
> 

Did you use separate validation dataset in both cases for mAP calculation, or did you use training dataset to check the mAP?
Did you measure AP50 in keras-yolov3?

**Comment by AdamCuellar** [comment]
@AlexeyAB A part of the validation set is similar but I’m using the whole dataset now so there’s more images. 

I measured AP50 with keras-yolov3. I can train and test on the subset with this repo to see how it does if you recommend that.
